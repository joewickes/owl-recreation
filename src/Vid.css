.Vid {
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  background-color: #555;
  flex: 1;
}

.Vid-box {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
}

.Vid-box-back {
  background-color: #000;
  width: 50%;
  display: flex;
  flex-direction: column;
  align-items: center;
  color: white;
}

.Vid-display img {
  background-color: #FFF;
  border-radius: 50%;
}

.Vid-box-name {
  font-weight: bold;
  margin-bottom: 5px;
  margin-top: 10px;
}

.Vid-box-line {
  width: 80%;
  height: 3px;
  background-color: red;
  border-radius: 5px;
  margin-bottom: 5px;
}

.Vid-box-icons {
  display: flex;
  justify-content: space-evenly;
  width: 100%;
  margin-bottom: 10px;
}