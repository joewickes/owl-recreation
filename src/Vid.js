import { render } from '@testing-library/react';
import React from 'react';
import './Vid.css';

class Vid extends React.Component {
  static defaultProps = {
    name: '',
    avatar: '',
    onStage: false
  };

  render() {
    const {name, avatar, onStage} = this.props;

    let vidDisplay = null;
    // if on stage, show them, else, don't
    if (onStage) {
      vidDisplay = (
        <div className="Vid">
          <div className="Vid-box">
            <div className="Vid-box-back">
              <p className="Vid-box-name">{name}</p>
              <div className="Vid-box-line"></div>
              <div className="Vid-box-icons">
                <p>Mic</p>
                <p>Mag</p>
              </div>
            </div>
          </div>
          <div className="Vid-display">
            <img src={avatar} alt={`${name}'s Avatar`} />
          </div>
        </div>
      );
    }

    return (
      vidDisplay
    );
  };
}

export default Vid;