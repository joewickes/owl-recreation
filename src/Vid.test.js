import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Vid from './Vid';

describe('Vid Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Vid />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the Vid on stage', () => {
    const tree = renderer
      .create(<Vid name="On Stage" avatar={'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1'} inSession={true} onStage={true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders empty div if in session but not on stage', () => {
    const tree = renderer
      .create(<Vid name="In Session Only" avatar={'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1'} inSession={true} onStage={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });


  it('renders empty div if not on stage or in session', () => {
    const tree = renderer
      .create(<Vid name="left" avatar={'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1'} inSession={false} onStage={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});