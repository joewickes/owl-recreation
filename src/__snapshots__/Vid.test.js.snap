// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vid Component renders empty div if in session but not on stage 1`] = `null`;

exports[`Vid Component renders empty div if not on stage or in session 1`] = `null`;

exports[`Vid Component renders the Vid on stage 1`] = `
<div
  className="Vid"
>
  <div
    className="Vid-box"
  >
    <div
      className="Vid-box-back"
    >
      <p
        className="Vid-box-name"
      >
        On Stage
      </p>
      <div
        className="Vid-box-line"
      />
      <div
        className="Vid-box-icons"
      >
        <p>
          Mic
        </p>
        <p>
          Mag
        </p>
      </div>
    </div>
  </div>
  <div
    className="Vid-display"
  >
    <img
      alt="On Stage's Avatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
  </div>
</div>
`;
